packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -363,7 +363,7 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
             const depModule = await moduleGraph.ensureEntryFromUrl(
               url,
               ssr,
-              canSkipImportAnalysis(url)
+              canSkipImportAnalysis(removeImportQuery(url))
```

Comment:
Reordering doesn't seem possible as we need the `url` here too.
I think that `canSkipImportAnalysis` should be able to cope with the `?import` query so we should deal with it inside the function.
Maybe we should instead modify `skipRE`?
```js
const skipRE = /\.(map|json)($|\\?)/
```

Reply:
That looks nice. I've changed to that.

BTW `/\\?/` won't work as you expect, I think.
```js
/a\\?b/.test('ab') // true
/a\\?b/.test('a?b') // false

/a\?b/.test('a?b') // true
/a\?b/.test('ab') // false
```
I found some of these in the code base so I'll create a PR to fix them.

Reply:
Ah, ya, sorry, I copy and pasted from one that places 😄